server/controllers/request.js

```diff
@@ -0,0 +1,69 @@
+const Request = require("../models/Request");
+const asyncHandler = require("express-async-handler");
+
+// @route GET /requests
+// @desc list of requests for logged in user
+// @access Private
+exports.getRequests = asyncHandler(async (req, res, next) => {
+  const user = req.body.userId;
+
+  let requests;
+  if (user) {
+    requests = await Request.find({
+      $or: [{ userId: user }, { sitterId: user }],
+    });
+  }
+
+  if (!user) {
+    res.status(404);
+    throw new Error("No quest found.");
+  }
+
+  res.status(200).json({ requests: requests });
+});
+
+// @route POST /requests
+// @desc Create a new request
+// @access Private
+exports.createRequest = asyncHandler(async (req, res, next) => {
+  const { userId, sitterId, start, end, timeZone } = req.body;
+
+  if (!userId || !sitterId || !start || !end || !timeZone) {
+    res.status(404);
```

Comment:
I would do a 400 error here as it's a bad request to process, also, in the future I would consider doing middleware validation.